src/coreclr/jit/lower.cpp

```diff
@@ -7396,10 +7397,14 @@ void Lowering::LowerSIMD(GenTreeSIMD* simdNode)
                 BlockRange().Remove(arg);
             }
 
-            assert(sizeof(constArgValues) == 16);
+            // For SIMD12, even though there might be 12 bytes of constants, we need to store 16 bytes of data
+            // since we've bashed the node the TYP_SIMD16 and do a 16-byte indirection.
+            assert(varTypeIsSIMD(simdNode));
+            const unsigned cnsSize = genTypeSize(simdNode);
+            assert(cnsSize <= sizeof(constArgValues));
 
-            unsigned cnsSize  = sizeof(constArgValues);
-            unsigned cnsAlign = (comp->compCodeOpt() != Compiler::SMALL_CODE) ? cnsSize : 1;
+            const unsigned cnsAlign =
+                (comp->compCodeOpt() != Compiler::SMALL_CODE) ? cnsSize : emitter::dataSection::MIN_DATA_ALIGN;
```

Comment:
I peeked at `emitSimd8Const`, `emitSimd16Const`, and `emitSimd32Const` and they all look to be doing the right thing here, but it might be good for someone else to double check.

This change looks good/correct to me, but it would probably be better if we lowered this to `GT_CNS_VEC` and went through the new normalized code path. Longer term, it would be good if we could get rid of the last 2-3 usages of `SIMDIntrinsicInitN` and replace them with `SimdAsHWIntrinsic` nodes instead.

Reply:
> but it might be good for someone else to double check.

Looks fine to me.